widget/entry_validation.go

```diff
@@ -0,0 +1,85 @@
+package widget
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/internal/widget"
+	"fyne.io/fyne/theme"
+)
+
+// SetValidationError manually updates the validation status until the next input change
+func (e *Entry) SetValidationError(err error) {
+	e.validationError = err
+
+	if e.Validator != nil {
+		e.validationStatus.Refresh()
+	}
+}
+
+var erroricon = theme.NewErrorThemedResource(theme.ErrorIcon()) // Avoid having to parse XML on each status refresh
```

Comment:
Note: this will not update if the theme changes

Reply:
Moving this to the CreateRenderer should not be too slow as the XML parse and render phase result is cached using the logic in internal/painter/svg_cache.go.

Reply:
I will just remove this step entirely and instead quote the wise @andydotxyz, by saying "Best to leave optimisations until we see that they are needed" :wink: